web-admin/src/features/projects/status/ActionsCell.svelte

```diff
@@ -0,0 +1,69 @@
+<script lang="ts">
+  import IconButton from "@rilldata/web-common/components/button/IconButton.svelte";
+  import * as DropdownMenu from "@rilldata/web-common/components/dropdown-menu";
+  import ThreeDot from "@rilldata/web-common/components/icons/ThreeDot.svelte";
+  import { createRuntimeServiceCreateTrigger } from "@rilldata/web-common/runtime-client";
+  import { RefreshCcwIcon } from "lucide-svelte";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import RefreshResourceConfirmDialog from "./RefreshResourceConfirmDialog.svelte";
+
+  export let resourceKind: string;
+  export let resourceName: string;
+  export let canRefresh: boolean;
+  export let triggerRefresh: (resourceName: string) => void;
+
+  let isConfirmDialogOpen = false;
+  let isDropdownOpen = false;
+
+  const createTrigger = createRuntimeServiceCreateTrigger();
+
+  async function refresh(resourceKind: string, resourceName: string) {
+    // First trigger the refresh
+    await $createTrigger.mutateAsync({
+      instanceId: $runtime.instanceId,
+      data: {
+        resources: [
+          {
+            kind: resourceKind,
+            name: resourceName,
+          },
+        ],
+      },
+    });
+
+    // Then update UI state and invalidate query
+    triggerRefresh(resourceName);
+  }
```

Comment:
It's a bit hard-to-follow that the refresh action has logic in 3 different places: this function, the inline function that starts on line 65, and the `triggerRefresh` callback function defined in `ProjectResources.svelte`. I'd propose unifying the functions in one place.

Reply:
This was needed to pass the resourceName upstream to display all the notifications in ProjectResources. I'll unify the functions.